src/service/viewer-impl.js

```diff
@@ -872,8 +872,16 @@ export class Viewer {
       return Promise.resolve('');
     }
     return this.sendMessageUnreliable_('fragment', undefined, true).then(
-      hash => hash || ''
-    );
+        hash => {
+          if (!hash) {
+            return '';
+          }
+          /* Strip leading '#' */
+          if (hash.indexOf('#') == 0) {
```

Comment:
> We should not have such defensive code if we don't need it.

I'd guarantee if we go the no-leading-`#` route, someone is going to just feed it `location.hash`. Then they get silent failures that'll be very difficult to track down.

> I prefer viewer response with no leading `#`

I like the other way, it simplifies integration code to leave the leading `#`.